actions/setup-token/action.yml

```diff
@@ -0,0 +1,98 @@
+name: setup token
+
+description: 'setup token'
+
+inputs:
+  slsa-workflow-recipient:
+    description: > 
+      The workflow filename that this token is intended for.
+
+      Example: delegator_generic_slsa3.yml
+    type: string
+    required: true
+
+  slsa-private-repository:
+    description: "If true, private repositories can post to the public transparency log."
+    required: false
+    type: boolean
+    default: false
+
+  slsa-build-artifacts-action-path:
+    description: >
+      The action path to invoke, from the root of the repository.
+      
+      Example: ./actions/build-artifacts'
+    type: string
+    required: true
+
+  slsa-runner-label:
+    description: >
+      The runner label to run the callback Action (`slsa-build-artifacts-action-path`) on.
+    type: choice
+    options: 
+        - ubuntu-latest
+    required: true
+
+  slsa-workflow-inputs:
+    description: >
+      The inputs to the Tool Reusable Workflow (TRW). The inputs
+      will be recorded in the provenance as the builder's inputs.
+
+      Note: The TRW is the reusable workflow calling this Action.
+    type: string
+    required: true
+
+outputs:
+  slsa-token:
+    description: 'SLSA token'
+    value: ${{ steps.sign.outputs.slsa-signed-token }}
+
+runs:
+  using: 'composite'
+  steps:
+      - name: Checkout the builder
+        uses: slsa-framework/slsa-github-generator/.github/actions/secure-builder-checkout@main
+        with:
+          # NOTE: These are always the same as the `@ref` above.
+          repository: ${{ github.action_repository }}
+          ref: ${{ github.action_ref }}
+
+      - uses: actions/setup-node@8c91899e586c5b171469028077307d293428b516 # tag=v3.5.1
+        with:
+          node-version: 16
+
+      - name: Debug
+        shell: bash
+        env:
+          INPUTS: ${{ toJson(inputs) }}
+          WORKFLOW_INPUTS: ${{ toJson(fromJson(inputs.slsa-workflow-inputs)) }}
+        run: |
+          echo "inputs=$INPUTS"
+          echo "workflow-inputs=$WORKFLOW_INPUTS"
+
+      - name: Generate unsigned token
+        id: generate
+        shell: bash
+        working-directory: ./__BUILDER_CHECKOUT_DIR__/actions/setup-token
+        env:
+          ACTION_INPUTS: ${{ toJson(inputs) }}
+        run: ./bin/setup-token.js
+      
+      - name: Debug
+        shell: bash
+        env:
+          REPO: ${{ github.action_repository }}
+          REF: ${{ github.action_ref }}
+        run: |
+          echo "repo: $REPO"
+          echo "ref: $REF"
+          echo "token: ${{ steps.generate.outputs.base64-slsa-token }}"
+          echo "recipient: ${{ inputs.slsa-workflow-recipient }}"
+
+      - name: Sign the token
+        id: sign
+        bash: shell
+        TODO: enable when the sign-token Action is merged.
+        uses: ./__BUILDER_CHECKOUT_DIR__/.github/actions/sign-token
```

Comment:
we can't do this if we keep the code for generating the unsigned token in this Action, because it would have to call itself and would run into an infinite loop. What I'll do is move the unsigned token generation to an internal Action and have the setup-token Action just do:
```shell
uses: internal/generate-unsigned-token@main
uses: internal/sign-token@main
```

This should take care of all the other comments as well (use of env variables instead of Action inputs, etc) and make the code cleaner.

Reply:
How important is it to keep these separate? From what I can tell the `sign-token` action is only just filling in a JSON object and base64 encoding it so it's super simple, so I don't think we need to separate due to complexity or testability.

Do we have use cases where we want to call just one or the other? how useful is an unsigned token?

Reply:
the signing Action will be shared for other uses cases like attestation signing. But now that you mention it, I think the signing format may be different and we finally found out how to use their API (which we had not initially), so it may be feasible to have it all in one Action.

@asraa I think we could use https://github.com/sigstore/sigstore-js/blob/825f81ca3d4453cf1252f08d63ee6a164eca9b40/src/sign.ts#L45 directly, correct?

Reply:
It's probably better to invoke the signing internally, since the other signing we're most interested in is attestation signing which has a different API.

Yeah! You can use it directly in the code.

like this, but not signAttestation, just sign blob: https://cs.github.com/laurentsimon/slsa-delegator/blob/2a0a15ed01341dd8b2dee1b7eddd901db2f87947/.github/actions/internal/sign-attestation/src/main.ts?q=sigstore.sigstore.sign#L30

Reply:
great, that's the one I will use. Thanks for confirming.